exporter/sumologicexporter/sender.go

```diff
@@ -0,0 +1,277 @@
+			}
+		}
+
+		// If data was sent, cleanup the currentTimeSeries counter
+		if sent {
+			currentRecords = currentRecords[:0]
+		}
+
+		// If log has been appended to body, increment the currentTimeSeries
+		if appended {
+			currentRecords = append(currentRecords, record)
+		}
+	}
+
+	if err := s.send(LogsPipeline, strings.NewReader(body.String()), flds); err != nil {
+		errs = append(errs, err)
+		droppedRecords = append(droppedRecords, currentRecords...)
+	}
+
+	if len(errs) > 0 {
+		return droppedRecords, componenterror.CombineErrors(errs)
+	}
+	return droppedRecords, nil
+}
+
+// sendLogsJSONFormat sends log records from the buffer as json data
+// and as the result of execution returns array of records which has not been
+// sent correctly and error
+func (s *sender) sendLogsJSONFormat(flds fields) ([]pdata.LogRecord, error) {
+	var (
+		body strings.Builder
+		errs []error
+		// droppedRecords tracks all dropped records
+		droppedRecords []pdata.LogRecord
+		// currentRecords tracks records which are being actually processed
+		currentRecords []pdata.LogRecord
+	)
+
+	for _, record := range s.buffer {
+		data := s.filter.filterOut(record.Attributes())
+		data[logKey] = record.Body().StringVal()
+
+		nextLine, err := json.Marshal(data)
+		if err != nil {
+			droppedRecords = append(droppedRecords, record)
+			errs = append(errs, err)
+			continue
+		}
+
+		sent, appended, err := s.appendAndSend(bytes.NewBuffer(nextLine).String(), LogsPipeline, &body, flds)
+		if err != nil {
+			errs = append(errs, err)
+			if sent {
+				droppedRecords = append(droppedRecords, currentRecords...)
+			}
+
+			if !appended {
+				droppedRecords = append(droppedRecords, record)
+			}
+		}
+
+		// If data was sent, cleanup the currentRecords
+		if sent {
+			currentRecords = currentRecords[:0]
+		}
+
+		// If log has been appended to body, add it to the currentRecords
+		if appended {
+			currentRecords = append(currentRecords, record)
+		}
+	}
+
+	if err := s.send(LogsPipeline, strings.NewReader(body.String()), flds); err != nil {
+		errs = append(errs, err)
+		droppedRecords = append(droppedRecords, currentRecords...)
+	}
+
+	if len(errs) > 0 {
+		return droppedRecords, componenterror.CombineErrors(errs)
+	}
+	return droppedRecords, nil
+}
+
+// appendAndSend appends line to the request body that will be sent and sends
+// the accumulated data if the internal buffer has been filled (with maxBufferSize elements).
+// It returns 2 booleans saying whether the data was sent and whether the provided data
+// was appended and an error if anything failed.
+func (s *sender) appendAndSend(
+	line string,
+	pipeline PipelineType,
+	body *strings.Builder,
+	flds fields,
+) (sent bool, appended bool, _ error) {
+	var errors []error
+	// sent gives information if the data was sent or not
+	sent = false
+	// appended keeps state of appending new log line to the body
+	appended = true
+
+	if body.Len() > 0 && body.Len()+len(line) > s.config.MaxRequestBodySize {
```

Comment:
```suggestion
	if body.Len() > 0 && body.Len()+len(line) >= s.config.MaxRequestBodySize {
```

You probably want to dispatch it if the new body size is equal to the max ;-)